src/main/java/org/cbio/gdcpipeline/listener/MutationStepListener.java

```diff
@@ -0,0 +1,105 @@
+package org.cbio.gdcpipeline.listener;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.cbio.gdcpipeline.model.rest.response.Hits;
+import org.cbio.gdcpipeline.util.CommonDataUtil;
+import org.springframework.batch.core.ExitStatus;
+import org.springframework.batch.core.StepExecution;
+import org.springframework.batch.core.StepExecutionListener;
+import org.springframework.beans.factory.annotation.Value;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Dixit Patel
+ **/
+public class MutationStepListener implements StepExecutionListener {
+    private static Log LOG = LogFactory.getLog(MutationStepListener.class);
+
+    @Value("${mutation.data.file.prefix}")
+    private String MUTATION_DATA_FILE_PREFIX;
+
+    @Value("${mutation.default.merged.maf.file}")
+    private String MUTATION_DEFAULT_MERGED_MAF_FILE;
+
+    @Value("#{jobParameters[sourceDirectory]}")
+    private String sourceDir;
+
+    @Value("#{jobParameters[outputDirectory]}")
+    private String outputDir;
+
+    @Value("#{jobExecutionContext[gdcFileMetadatas]}")
+    private List<Hits> gdcFileMetadatas;
+
+    @Value("#{jobParameters[separate_maf]}")
+    private String separate_maf;
+
+    @Override
+    public void beforeStep(StepExecution stepExecution) {
+        boolean isStarted = false;
+        if (stepExecution.getJobExecution().getExecutionContext().containsKey("isStartedMaf")) {
+            isStarted = true;
+        }
+        if (!isStarted) {
```

Comment:
I don't understand why this `isStarted` variable is needed, and the logic surrounding it. Why can't you just have a list, put mafs in it that should be processed, and that be it? You can clear the list out of the execution context every time `beforeStep` is called if the issue is having a clean list, and just pop out the one from the list of available mafs as you put it into the list you want to have processed. Am I missing something?

Reply:
Done.